### PR TITLE
fixes up some helmets

### LINKED ISSUE #14904: You can't adjust the riot helmet face shield while buckled to a chair.
Not sure if this is a bug, but I don't know why it would be intended.

### PR DESCRIPTION
**Pull request by PKPenguin321**
i honestly do not know what i am doing any more i am just parroting what @neersighted said to do
fixes https://github.com/tgstation/-tg-station/issues/14904 maybe

### CODE REVIEW

**Comment by PKPenguin321** [comment]
SHIT
@neersighted help
ill fix stuff when i get up tomorrow because it's LATE AS SHIT here and i'm tired but in the mean time you can review code

**Comment by optimumtact** [comment]
you need to define the user, iirc it's passed to attack_self as a parameter

File: code/modules/clothing/head/helmet.dm [hunk 1]
```diff
@@ -49,7 +49,7 @@
 	flags_cover = HEADCOVERSEYES | HEADCOVERSMOUTH
 
 /obj/item/clothing/head/helmet/attack_self()
```
Comment (neersighted) [inline code/modules/clothing/head/helmet.dm]: should accept mob/user

File: code/modules/clothing/head/helmet.dm [hunk 2]
```diff
@@ -57,11 +57,11 @@
...
 			flags_cover ^= initial(flags_cover)
 			icon_state = "[initial(icon_state)][up ? "up" : ""]"
-			usr << "[up ? alt_toggle_message : toggle_message] \the [src]"
+			user << "[up ? alt_toggle_message : toggle_message] \the [src]"
 
-			usr.update_inv_head()
-			if(istype(usr, /mob/living/carbon))
-				var/mob/living/carbon/C = usr
+			user.update_inv_head()
+			if(istype(user, /mob/living/carbon))
```
Comment (neersighted) [inline code/modules/clothing/head/helmet.dm]: iscarbon

File: code/modules/clothing/head/helmet.dm [hunk 3]
```diff
@@ -49,7 +49,7 @@
 	flags_cover = HEADCOVERSEYES | HEADCOVERSMOUTH
 
 /obj/item/clothing/head/helmet/attack_self()
-	if(usr.canmove && !usr.stat && !usr.restrained() && can_toggle)
+	if(can_toggle && !user.incapacitated())
 		if(world.time > cooldown + toggle_cooldown)
 			cooldown = world.time
 			up ^= 1
```
Comment (neersighted) [inline code/modules/clothing/head/helmet.dm]: TRUE

Comment (duncathan) [inline code/modules/clothing/head/helmet.dm]: If `up` is Boolean then just replace this line with `up = !up`

Comment (neersighted) [inline code/modules/clothing/head/helmet.dm]: Yeah, what dunc said.

File: code/modules/clothing/head/helmet.dm [hunk 4]
```diff
@@ -57,11 +57,11 @@
...
 
-			usr.update_inv_head()
-			if(istype(usr, /mob/living/carbon))
-				var/mob/living/carbon/C = usr
+			user.update_inv_head()
+			if(istype(user, /mob/living/carbon))
+				var/mob/living/carbon/C = user
 				C.head_update(src, forced = 1)
 
 			if(active_sound)
```
Comment (neersighted) [inline code/modules/clothing/head/helmet.dm]: If you can improve this code that would be great.

**Comment by PKPenguin321** [comment]
god damn jesus shit fuck helmet code is my ass

**Comment by neersighted** [comment]
Its old code... *shrug*

File: code/modules/clothing/head/helmet.dm (hunk 4 above)
Comment (PKPenguin321) [inline code/modules/clothing/head/helmet.dm]: i have no idea what it's even trying to do

**Comment by optimumtact** [comment]
Looks good now :+1:

**Comment by neersighted** [comment]
:+1: merge it

**Comment by tkdrg** [comment]
I cleaned this up just the other day why do you hate xor
I don't really care though

**Pull request merged**